nats-spring-cloud-stream-binder/src/test/java/io/nats/cloud/stream/binder/BinderTests.java

```diff
@@ -404,4 +403,63 @@ public void testMessageHandler() throws IOException, InterruptedException {
+
+                final String request = "hello request";
+                final String reply = "hello reply";
+                final String req2rep = "req2rep";
+                final String rep2req = "rep2req";
+                final MessageHandler appOneMh = binder.createProducerMessageHandler(provisioner.provisionProducerDestination(req2rep, null), null, null);
+                final CompletableFuture<org.springframework.messaging.Message<?>> appOneReceived = new CompletableFuture<>();
+                final NatsMessageProducer appOneMp = buildMessageProducer(rep2req, appOneReceived, binder, provisioner);
+                appOneMp.start();
+
+                final MessageHandler appTwoMh = binder.createProducerMessageHandler(provisioner.provisionProducerDestination(rep2req, null), null, null);
+                final CompletableFuture<org.springframework.messaging.Message<?>> appTwoReceived = new CompletableFuture<>();
+                final NatsMessageProducer appTwoMp = buildMessageProducer(req2rep, appTwoReceived, binder, provisioner);
+                appTwoReceived.thenAccept(requestMsg -> {
+                    // appTwo replies copying headers
+                    appTwoMh.handleMessage(MessageBuilder.withPayload(reply + " to " + new String((byte[]) requestMsg.getPayload(), UTF_8)).copyHeaders(requestMsg.getHeaders()).build());
+                });
+                appTwoMp.start();
+
+                // appOne sends request
+                appOneMh.handleMessage(MessageBuilder.withPayload(request).setHeader(MessageHeaders.REPLY_CHANNEL, rep2req).build());
+
+                // appTwo receives request with header (and replies)
+                final org.springframework.messaging.Message<?> appTwoMessage = appTwoReceived.get(5, TimeUnit.SECONDS);
+                assertEquals(rep2req, appTwoMessage.getHeaders().get(MessageHeaders.REPLY_CHANNEL));
+                assertEquals(request, new String((byte[]) appTwoMessage.getPayload(), UTF_8));
+
+                // appOne receives the reply
+                final org.springframework.messaging.Message<?> appOneMessage = appOneReceived.get(5, TimeUnit.SECONDS);
+                assertEquals(rep2req, appOneMessage.getHeaders().get(MessageHeaders.REPLY_CHANNEL));
+                assertEquals(reply + " to " + request, new String((byte[]) appOneMessage.getPayload(), UTF_8));
+            });
+        }
+    }
+
+    private NatsMessageProducer buildMessageProducer(String subject, CompletableFuture<org.springframework.messaging.Message<?>> received, NatsChannelBinder binder, NatsChannelProvisioner provisioner) {
+        final NatsMessageProducer mp = (NatsMessageProducer) binder.createConsumerEndpoint(provisioner.provisionConsumerDestination(subject, "", null), "", null);
+        final DirectChannel output = new DirectChannel();
+        output.subscribe(received::complete);
+        mp.setOutputChannel(output);
+        return mp;
+    }
```

Comment:
Please add newline to end

Reply:
No other test file does that but sure, no problem.
Again, a `.editorcongig` would help keep everything in check.